compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/InteractionUIView.uikit.kt

```diff
@@ -39,52 +50,354 @@ internal enum class CupertinoTouchesPhase {
+     * Checks whether the centroid location of [trackedTouches] has exceeded the scrolling slop
+     * relative to [initialLocation]
+     */
+    private val isLocationDeltaAboveSlope: Boolean
+        get() {
+            val initialLocation = initialLocation ?: return false
+            val centroidLocation = trackedTouchesCentroidLocation ?: return false
+
+            val slop = 10.0
+
+            val dx = centroidLocation.useContents { x - initialLocation.useContents { x } }
+            val dy = centroidLocation.useContents { y - initialLocation.useContents { y } }
+
+            return dx * dx + dy * dy > slop * slop
+        }
+
+    /**
+     * Calculates the centroid of the tracked touches.
+     */
+    private val trackedTouchesCentroidLocation: CValue<CGPoint>?
+        get() {
+            if (trackedTouches.isEmpty()) {
+                return null
+            }
+
+            var centroidX = 0.0
+            var centroidY = 0.0
+
+            for (touch in trackedTouches) {
+                val location = touch.locationInView(view)
+                location.useContents {
+                    centroidX += x
+                    centroidY += y
+                }
+            }
+
+            return CGPointMake(
+                x = centroidX / trackedTouches.size.toDouble(),
+                y = centroidY / trackedTouches.size.toDouble()
+            )
+        }
+
+    /**
+     * Implementation of [CMPGestureRecognizerHandlerProtocol] that handles touchesBegan event and
+     * forwards it here.
+     *
+     * There are following scenarios:
+     * 1. Those are first touches in the sequence, the interaction view is hit-tested. In this case, we
+     * should start the gesture recognizer immediately and start passing touches to the Compose
+     * runtime.
+     *
+     * 2. Those are first touches in the sequence, an interop view is hit-tested. In this case we
+     * intercept touches from interop views until the gesture recognizer is explicitly failed.
+     * See [UIGestureRecognizer.delaysTouchesBegan]. In the same time we schedule a failure in
+     * [CMPGestureRecognizer.scheduleFailure], which will pass intercepted touches to the interop
+     * views if the gesture recognizer is not recognized within a certain time frame
+     * (UIScrollView reverse-engineered 150ms is used).
+     * The similar approach is used by [UIScrollView](https://developer.apple.com/documentation/uikit/uiscrollview?language=objc)
+     *
+     * 3. Those are not the first touches in the sequence. A gesture is recognized.
+     * We should continue with scenario (1), we don't yet support multitouch sequence in
+     * compose and interop view simultaneously (e.g. scrolling native horizontal
+     * scroll and compose horizontal scroll with different fingers)
+     *
+     * 4. Those are not the first touches in the sequence. A gesture is not recognized.
+     * See if centroid of the tracked touches has moved enough to recognize the gesture.
```

Comment:
If second touch lands after the gesture was failed, it's not even delivered to it, so the gesture will properly be recognised by maps, for example.

Reply:
I'm afraid this might break the behavior of detectTransformGestures. On 1.7.0-alpha02, it seemed detectTransformGestures always call onGesture with zoom = 1.0f on iOS.
I have create an issue on youtrack: https://youtrack.jetbrains.com/issue/CMP-5879/detectTransformGestures-doesnt-work-properly-on-iOS-with-Compose-Multiplatform-1.7.0-alpha02

Reply:
Hi, thanks, already looking into it.